exotica/src/Scene.cpp

```diff
@@ -840,9 +845,87 @@ namespace exotica
   void Scene::cleanScene()
   {
       getPlanningScene()->removeAllCollisionObjects();
+      updateSceneFrames();
       collision_scene_->reinitializeCollisionWorld();
   }
 
+  void Scene::updateSceneFrames()
+  {
+      kinematica_.resetModel();
+
+      planning_scene::PlanningScenePtr ps = collision_scene_->getPlanningScene();
+
+      // Add world objects
+      for(auto& object : *ps->getWorld())
+      {
+          if(object.second->shapes_.size())
+          {
+              // Use the first collision shape as the origin of the object
+              Eigen::Affine3d objTransform = object.second->shape_poses_[0];
+              std::string objectName = object.first;
+              kinematica_.AddElement(objectName, objTransform);
+              for(int i=0; i<object.second->shape_poses_.size();i++)
+              {
+                  Eigen::Affine3d trans = objTransform.inverse()*object.second->shape_poses_[i];
+                  kinematica_.AddElement(objectName+"_collision_"+std::to_string(i), trans, objectName, object.second->shapes_[i]);
+              }
+          }
+          else
+          {
+            HIGHLIGHT("Object with no shapes ('"<<object.first<<"')");
+          }
+      }
+
+      // Add robot collision objects
+      ps->getCurrentStateNonConst().update(true);
+      const std::vector<const robot_model::LinkModel*>& links =
+          ps->getCollisionRobot()->getRobotModel()->getLinkModelsWithCollisionGeometry();
+      for (int i = 0; i < links.size(); ++i)
+      {
+          Eigen::Affine3d objTransform = ps->getCurrentState().getGlobalLinkTransform(links[i]);
+
+          for (int j = 0; j < links[i]->getShapes().size(); ++j)
+          {
+              Eigen::Affine3d trans = objTransform.inverse()*ps->getCurrentState().getCollisionBodyTransform(links[i], j);
+              kinematica_.AddElement(links[i]->getName()+"_collision_"+std::to_string(j), trans, links[i]->getName(), links[i]->getShapes()[j]);
+          }
+      }
+
+      kinematica_.UpdateModel();
+  }
+
+  void Scene::attachObject(const std::string& name, const std::string& parent)
+  {
+      kinematica_.changeParent(name, parent, KDL::Frame(), false);
```

Comment:
I think the two overloads could be realised without code duplication. Just call one from the other passing in KDL::Frame() - in case the latter changes in the future

Reply:
I wanted different behaviour here. First overload attaches object from its current wold position, the second specifies the offset relative to the new parent. I will probably rename the methods to make this clearer.

Reply:
A doxygen to explain that difference may be fine and sufficient.